visualization/app/codeCharta/util/scenarioHelper.ts

```diff
@@ -49,14 +49,24 @@ export class ScenarioHelper {
 		return scenarioItems
 	}
 
-	private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: NodeMetricData[]) {
-		if (scenario.area && !metricData.find(x => x.name === scenario.area.areaMetric)) {
-			return false
-		}
-		if (scenario.color && !metricData.find(x => x.name === scenario.color.colorMetric)) {
-			return false
+	private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: MetricData) {
+		const { area, color, height, edge } = scenario
+
+		if (area || color || height) {
+			const nodeMetricSet = new Set(metricData.nodeMetricData.map(data => data.name))
+
+			if (area && !nodeMetricSet.has(area.areaMetric)) {
+				return false
+			}
+			if (color && !nodeMetricSet.has(color.colorMetric)) {
+				return false
+			}
+			if (height && !nodeMetricSet.has(height.heightMetric)) {
+				return false
+			}
 		}
-		if (scenario.height && !metricData.find(x => x.name === scenario.height.heightMetric)) {
+
+		if (edge && !metricData.edgeMetricData.find(x => x.name === edge.edgeMetric)) {
 			return false
 		}
 
```

Comment:
Just out of curiosity:

We could remove the last return true and include it to the edge query. It would look like this

`return !(edge && !metricData.edgeMetricData.find(x => x.name === edge.edgeMetric))`

but I don't know if the readability got worse. How would you guys handle that?

Reply:
I like the shorter notation. As long as we don't have any other cases, I'd try to keep the code as short and readable as possible.